### DISCUSSION TITLE
How is the ROTATE_DOOR option used?

### DISCUSSION THREAD

**Discussion by djkraven** [mrlt8/docker-wyze-bridge#159] [Q&A]
I am using the docker-compose.yml file and tried True, 90, yes as options but it does not rotate the stream in the browser. I must be missing something simple. [oddly the code block is not taking the line feeds] This is with the Cam v3 is the doorbell the only support cam for rotate or could the other cam be enabled?

`version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest # Use a prebuilt image
        # build:  # Uncomment to build from source
        #     context: ./app # Uncomment to build from source
        #     # dockerfile: Dockerfile.arm # Uncomment to build for arm
        ports:
            - 1935:1935
            - 8554:8554
            - 8888:8888
        environment:
            - WYZE_EMAIL=
            - WYZE_PASSWORD=
            - ROTATE_DOOR=`

**Comment by mrlt8** [comment]
AFAIK, the only way to rotate the video is to re-encode which can be resource intensive, so ROTATE_DOOR will only rotate WYZEDB3 cameras.

You can always pass your own FFMPEG commands directly to the bridge with something like:
```
environment:
  - FFMPEG_CMD_MYCAMNAME= -i - -vcodec libx264 -vf transpose=1 -f rtsp rtsp://0.0.0.0:8554/
```

**Reply by djkraven** [reply]
Is the rotate re-encoded happening with the docker on my PC so the resource-intensive part is partly dependent on the PC hardware? I will have to look into your suggestion of the bridge. Ideally, I am looking to view it on a browser. Another option might be to use CSS formatting of a frame.

**Reply by mrlt8** [reply]
If you can rotate the video in the browser, that would probably be the best solution.

**Reply by djkraven** [reply]
This is a sample not perfect of how to rotate using iframe and CSS. Not sure it is possible to incorporate the transform option into the web page option of the Docker Wyze Bridge or if that is at the Wyze side of things.
[Wyze-Rotate.html.txt](https://github.com/mrlt8/docker-wyze-bridge/files/7348004/Wyze-Rotate.html.txt)
